src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs

```diff
@@ -344,42 +346,44 @@ public ProjectInfo WithAssemblyName(string assemblyName)
             return With(attributes: Attributes.With(assemblyName: assemblyName));
         }
 
-        public ProjectInfo WithOutputFilePath(string outputFilePath)
+        public ProjectInfo WithOutputFilePath(string? outputFilePath)
         {
             return With(attributes: Attributes.With(outputPath: outputFilePath));
         }
 
-        public ProjectInfo WithOutputRefFilePath(string outputRefFilePath)
+        public ProjectInfo WithOutputRefFilePath(string? outputRefFilePath)
         {
             return With(attributes: Attributes.With(outputRefPath: outputRefFilePath));
         }
 
-        public ProjectInfo WithDefaultNamespace(string defaultNamespace)
+        public ProjectInfo WithDefaultNamespace(string? defaultNamespace)
         {
             return With(attributes: Attributes.With(defaultNamespace: defaultNamespace));
         }
 
-        public ProjectInfo WithCompilationOptions(CompilationOptions compilationOptions)
+        public ProjectInfo WithCompilationOptions(CompilationOptions? compilationOptions)
         {
+            // The With method here doesn't correctly handle the null value, tracked by https://github.com/dotnet/roslyn/issues/37880
```

Comment:
💭 This comment seems unnecessary. I really wish it used a default value instead of null here since it's just accounting for F# and other extension languages.

Reply:
So this is more fun ProjectInfo allows null even for C#/VB. That means "when I create a project with this info, please use the default value".